client/state/console-dispatch/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Console dispatcher Redux store enhancer
+ *
+ * @blame dmsnell
+ *
+ * Inject into the `createStore` enhancer chain in order
+ * to provide access to the store directly from the console.
+ *
+ * Will only attach if the `window` variable is available
+ * globally. If not it will simply be an empty link in the
+ * chain, passing straight through.
+ *
+ * A few helpers have also been attached to the window in
+ * order to make debugging and interacting easier. Please
+ * see the README for more information.
+ *
+ */
+
+/**
+ * External dependencies
+ */
+import {
+	matchesProperty,
+	now,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import * as actionTypes from 'state/action-types';
+
+const state = {
+	actionHistory: [],
+	shouldRecordActions: true,
+	historySize: 100,
+};
+
+const actionLog = {
+	filter: type => state.actionHistory.filter( matchesProperty( 'type', type ) ),
+	setSize: size => state.historySize = size,
+	start: () => state.shouldRecordActions = true,
+	stop: () => state.shouldRecordActions = false,
+};
+
+Object.defineProperty( actionLog, 'history', {
+	enumerable: true,
+	get: () => state.actionHistory,
+} );
+
+const recordAction = action => {
+	const {
+		actionHistory,
+		historySize,
+	} = state;
+
+	actionHistory.push( { ...action, meta: { ...action.meta, timestamp: now() } } );
```

Comment:
Not a matter of need really. I don't think there are any real performance issues here. `Date.now()` does have one glaring issue for which I like to avoid it: it's not guaranteed to increase at a constant rate. It could go backwards if the system clock skews.